### PR TITLE
refactor: migrate to react-intl v3

### PR DESCRIPTION
**Pull request by korhaliv**
## Description:
Migrate to react-intl v3. In our case main breaking changes were:
Removal of `addLocaleData` 
Removal of `intlShape`
Removal of `FormattedRelative`

## How Has This Been Tested?
Manually 

## Types of changes:
Refactor

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have reviewed and updated the documentation accordingly.
- [x] I have read the _CONTRIBUTING_ document.
- [ ] I have added tests to cover my changes where needed.
- [ ] All new and existing tests passed.
- [x] My commits have been squashed into a concise set of changes.

### CODE REVIEW

File: i18n/intjShape.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import PropTypes from 'prop-types'
```
Comment (mrfelton) [inline i18n/intjShape.js]: The name of this file is `intjShape` - should probably be `intlShape`

File: renderer/components/Activity/ActivityActions/ActivityActions.js [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
 import PropTypes from 'prop-types'
-import { injectIntl, intlShape } from 'react-intl'
+import { injectIntl } from 'react-intl'
 import { Flex } from 'rebass'
 import { Card } from 'components/UI'
 import ActivityFilter from './ActivityFilter'
 import ActivityRefresh from './ActivityRefresh'
 import ActivitySearch from './ActivitySearch'
 import messages from './messages'
+import { intlShape } from '@zap/i18n'
```
Comment (mrfelton) [inline renderer/components/Activity/ActivityActions/ActivityActions.js]: import should go before relative imports (this comment applies to the whole PR)

File: i18n/index.js [hunk 3]
```diff
@@ -2,6 +2,9 @@ import get from 'lodash/get'
 import { lookup } from 'country-data-lookup'
 import createDebug from 'debug'
 import isRenderer from 'is-electron-renderer'
+import PropTypes from 'prop-types'
+
+export const intlShape = PropTypes.instanceOf(Object)
```
Comment (mrfelton) [inline i18n/index.js]: Why is this defined as a simple `Object` type here? Shouldn't it be `IntlShape` as per the seemingly unused `i18n/intjShape.js` file?

File: stories/components/countdown.stories.js [hunk 4]
```diff
@@ -8,22 +8,16 @@ storiesOf('Components', module).addWithChapters('Countdown', {
       sections: [
         {
           title: 'Default',
-          sectionFn: () => <Countdown date={Math.round(new Date() / 1000) + 5000} />,
+          sectionFn: () => <Countdown date={5} />,
```
Comment (mrfelton) [inline stories/components/countdown.stories.js]: `date={5}` seems quite non-explicit. 5 what? seconds? 5 months? I think if we want to change the way that this works then we should rename the prop to something that describes what is expected. `5` is not a `date` which is what I would expect to pass in to a prop named `date`

Comment (korhaliv) [reply stories/components/countdown.stories.js]: updated

File: i18n/index.js (hunk 3 above)
Comment (korhaliv) [reply i18n/index.js]: resolved

File: renderer/components/Activity/ActivityActions/ActivityActions.js (hunk 2 above)
Comment (korhaliv) [reply renderer/components/Activity/ActivityActions/ActivityActions.js]: updated

File: renderer/components/Request/RequestSummary.js [hunk 5]
```diff
@@ -1,52 +1,29 @@
...
-      schedulerInstance.addTask({
-        task: refreshIsExpired,
-        baseDelay: 1000,
-      })
-    }
-    // Clear scheduler on unmount.
+    setExpiryDelta(decodedInvoice.timeExpireDate - getUnixTime() / 1000)
     return () => {
-      schedulerInstance.removeAllTasks()
+      setIsExpired(false)
```
Comment (mrfelton) [inline renderer/components/Request/RequestSummary.js]: Why the need to update the state on unmount?

**Pull request merged**